lib/scope.js

```diff
@@ -1182,37 +1182,98 @@ module.exports = {
+
+    // da's header covers the top of the section containing the fields which means
+    // sometimes it blocks nodes from clicks. Scroll item to the bottom instead.
+    await handle.evaluate(( elem ) => { elem.scrollIntoView(false); return elem.scrollTop });
+    let set_to = row.value;
+    if ( type === `checkbox` ) { set_to = row.checked; }
+
+    if ( env_vars.DEBUG && handle ) {
+      let name = await handle.evaluate(( elem ) => {
+        try { elem.nextSibling.style.background = 'Khaki'; }
+        catch(err){ elem.style.background = 'Khaki'; }
+        return elem.getAttribute('for') || elem.getAttribute('name'); 
+      });
+      await scope.page.waitFor( 400 );
+    }
+
+    // Detect a custom type and get the element of that custom type if it exists
+    let datatype = await handle.evaluate(( elem )=> {
+      // If the element is in a form group* then extract the form group's
+      // datatype value.
+      // * For now the only custom datatype we have is in a form group. In
+      // future we should look into what happens with a `fieldset` for
+      // `choices:`, etc.
+      let custom_datatype = null, matches = null;
+      let group = elem.closest( `.form-group` );
+      if ( group && group.className ) {
+        matches = ` ${ group.className } `.match(/da-field-container-datatype-([^ ]+) /);
+      }
+      if ( matches ) {
+        custom_datatype = matches[1];
       }
 
+      return custom_datatype;
+    });
+
+    // Set field
+    if ( scope.setCustomDatatype[ datatype ] === undefined ) {
+      // If no custom datatype, set a regular field
       await scope.setField[ tag ]( scope, { handle, set_to });
+    } else {
+      // Set da package custom fields
+      let custom_handle = await handle.evaluateHandle(( elem )=> { return elem.closest( `.form-group` ); });
+      await scope.setCustomDatatype[ datatype ]( scope, { handle: custom_handle, set_to });
     }
 
     return !disabled;
   },  // Ends scope.setVariable()
 
+  // Handle setting values for da custom datatypes
+  // E.g. `da-field-container-datatype-BirthDate`
+  // TODO: Make this more easily extensible
+  setCustomDatatype: {
+    "BirthDate": async function ( scope, { handle, set_to }) {
+      /* Given a date of the format 'mm/dd/yyyy' and the handle of the
+      * custom field container, sets the ALToolbox custom birthdate
+      * field values. */
+      let date_parts = set_to.split(`/`);
+      let fields = await handle.$$(`select, input:not([type="hidden"])`);
+      for ( let field_i = 0; field_i < fields.length; field_i++ ) {
+        let field = fields[ field_i ];
+
+        if ( env_vars.DEBUG ) {
+          await field.evaluate( elem => { elem.style.background = `Khaki`; });
+        }
+
+        if ( field_i === 0 ) {
+          await field.select( date_parts[ 0 ] );
+        } else {
+          await scope.setText( scope, {
+            handle: field,
+            set_to: date_parts[ field_i ]
+          });
+        }
```

Comment:
Maybe add a comment here that the `if` statement sets months, and the `else` sets the date/year.

Reply:
Done!